Sources/PrivoSDK/internal/age-gate/AgeGateStorage.swift

```diff
@@ -76,14 +74,59 @@ internal class AgeGateStorage {
         return nil
     }
     
-    func getFpId() async -> String {
-        if let fpId = keychain.get(getFpIdKey()) { return fpId }
-        let fingerprint = DeviceFingerprint()
-        let response = await api.generateFingerprint(fingerprint: fingerprint)
-        //TO DO: need to implement for managing way when there is no id
-        guard let id = response?.id else { return "" }
-        keychain.set(key: getFpIdKey(), value: id)
-        return id
+    // MARK: FingerprintStorage
+    
+    var fpId: String? {
+        get {
+            return keychain.get(getFpIdKey())
+        }
+        set {
+            if let newValue {
+                keychain.set(key: getFpIdKey(), value: newValue)
+            } else {
+                keychain.delete(getFpIdKey())
+            }
+        }
+    }
+}
+
+protocol FpIdStorage {
+    var fpId: String? { get set }
+}
+
+protocol FpIdable {
+    var fpId: String? { get async }
+}
+
+class FpIdService: FpIdable {
```

Comment:
looks like we need to move it to a separate file. It's not a storage anymore

Reply:
moved it